### DISCUSSION TITLE
Conditional rendering - x-if and x-else

### DISCUSSION THREAD

**Discussion by claytsm** [alpinejs/alpine#1027] [4. General]
Is there such a thing as x-if AND x-else in Alpine.js?

**Comment by ryangjchandler** [comment]
@matteboy99 `x-else` is just the negation of the `x-if` condition so there's no real need to have it as part of the API.

**Reply by claytsm** [reply]
Ah thanks for that - I was wanting to show different html depending on a function returning true or false, keeping both types of html separate. The only work around I can have for this is having 2 templates calling my function basically twice when within a for loop... it seems that having the x-else would reduce the amount of calls.... or am I missing something? (previously loving Vue.js, so may have been spoilt).

**Reply by ryangjchandler** [reply]
Yeah, `x-else` would reduce the amount of calls by one so would only be a big performance benefit if the function being called is super expensive.

**Reply by claytsm** [reply]
Indeed - I'm using Alpine.js along with Shopify so it could get quite hungry... i'm looking at workarounds. Thanks for the comments :)
